pybamm_tea/tea.py

```diff
@@ -394,223 +425,191 @@ def print_capacities_and_potentials(self):
-            and pava.get("Positive electrode stoichiometry at 100%") is not None
-            and pava.get("Positive electrode stoichiometry at 0%") is not None
+            pava.get("Negative electrode stoichiometry at 0% SoC") is None
+            and pava.get("Negative electrode stoichiometry at 100% SoC") is None
+            and pava.get("Positive electrode stoichiometry at 100% SoC") is not None
+            and pava.get("Positive electrode stoichiometry at 0% SoC") is not None
         ):
             y100 = pava.get("Positive electrode stoichiometry at 100% SoC")
             y0 = pava.get("Positive electrode stoichiometry at 0% SoC")
-            # x0, x100 = self.get_stoichiometries(pava, y0, y100)
-            # update cut-off voltages if voltage curve(s) are provided
+            # calculate lithium in positive electrode at SOC = 0
+            n_pe_0 = y0 * (pava.get("Maximum concentration in positive electrode [mol.m-3]")
+                           * pava.get("Positive electrode thickness [m]")
+                           * pava.get("Positive electrode active material volume fraction")
+                           * 96485 / 3.6 / 10000)
+            # calculate lithium in positive electrode at SOC = 1
+            n_pe_100 = y100 * (pava.get("Maximum concentration in positive electrode [mol.m-3]")
+                           * pava.get("Positive electrode thickness [m]")
+                           * pava.get("Positive electrode active material volume fraction")
+                           * 96485 / 3.6 / 10000)
+            if n_pe_0 > pava.get("Lithium inventory [mA.h.cm-2]"):
+                raise ValueError("Error: Stoichiometry calculation failed. Lithium inventory in positive electrode at 0% SoC > Lithium inventory [mA.h.cm-2]")
+            # calculate lithium in negative electrode at SOC = 0
+            x0 = ((pava.get("Lithium inventory [mA.h.cm-2]") - n_pe_0)
+                      / (pava.get("Maximum concentration in negative electrode [mol.m-3]")
+                         * pava.get("Negative electrode thickness [m]")
+                         * pava.get("Negative electrode active material volume fraction")
+                         * 96485 / 3.6 / 10000))
+            # calculate lithium in negative electrode at SOC = 1
+            x100 = ((pava.get("Lithium inventory [mA.h.cm-2]") - n_pe_100)
+                      / (pava.get("Maximum concentration in negative electrode [mol.m-3]")
+                         * pava.get("Negative electrode thickness [m]")
+                         * pava.get("Negative electrode active material volume fraction")
+                         * 96485 / 3.6 / 10000))
+            if x0 < 0 or x100 < 0 or x0 > 1 or x100 > 1:
+                raise ValueError("Error: Stoichiometry calculation for negative electrode failed.")
+            if pava.get("Negative electrode OCP [V]") is not None and pava.get("Positive electrode OCP [V]") is not None:
```

Comment:
These voltage cut-offs are normally defined first for the cell, e.g. 2.5-4.2V, and then the stoichiometry windows are calculated to respect those voltage limits. This way seems backwards?